frame/system/src/lib.rs

```diff
@@ -1404,7 +1406,19 @@ impl<T: Trait + Send + Sync> CheckWeight<T> where
 				} else {
 					Ok(())
 				}
-			}
+			},
+			// For operational transactions we make sure it doesn't exceed
+			// the space alloted for `Operational` class.
+			DispatchClass::Operational => {
+				let maximum_weight = T::MaximumBlockWeight::get();
+				let operational_limit = Self::get_dispatch_limit_ratio(DispatchClass::Operational) * maximum_weight;
+				let extrinsic_weight = info.weight.saturating_add(T::ExtrinsicBaseWeight::get());
```

Comment:
This calculation is missing `BaseBlockWeight` if we want to be entirely accurate.

Reply:
What happens if a transaction satisfies this but can actually never be included because check_block_weight always fails ? Is that an issue like the transaction will stay forever in the queue ?
Because in the moment I guess it also doesn't include the minimum cost of OnInitialize.

Reply:
> Is that an issue like the transaction will stay forever in the queue ?

Yes, if it's immortal.

> Because in the moment I guess it also doesn't include the minimum cost of OnInitialize.

Good point. With `Normal` dispatchables, we have a separate parameter that allows to control the max size when runtime is created. I guess it might make sense to lift it up for `Operational` as well and define as like `90%` of max block weight or sth.